test/unittest/blas3/blas3_trsm_test.cpp

```diff
@@ -0,0 +1,114 @@
+
+#include "blas_test.hpp"
+
+template<typename scalar_t>
+using combination_t = std::tuple<int, int>;
+
+template <typename T>
+void transpose(T* A, size_t m, size_t n, size_t lda) {
+  for (size_t i = 0; i < m; ++i) {
+    for (size_t j = 0; j < i; ++j) {
+      std::swap(A[j + i * lda], A[i + j * lda]);
+    }
+  }
+}
+
+template <typename T>
+void printMatrix(T* A, size_t m, size_t n, size_t lda,
+                 bool bIsColMajor = true) {
+  for (size_t i = 0; i < m; ++i) {
+    for (size_t j = 0; j < n; ++j) {
+      T Element = bIsColMajor ? A[i + j * lda] : A[j + i * lda];
+      std::cout << std::setw(4) << Element << " ";
+    }
+    std::cout << std::endl;
+  }
+}
+
+template<typename scalar_t>
+void run_test(const combination_t<scalar_t> combi) {
+  int m;
+  int n;
+  std::tie(m, n) = combi;
+
+  using data_t = utils::data_storage_t<scalar_t>;
+
+  // clang-format off
+  std::vector<data_t> A = {
+    1, 0, 0, 0,
```

Comment:
It doesn't look like the input data will work with anything else than `combi == {4, 4}`. I think that's fine, but we should probably have a TODO to generate the matrix using a function or something.

Reply:
Already removed